File: two_word_games.js
#!/usr/bin/env node
var fs = require('fs');

function contains(list,element) {
  return list.filter(function(a) { return a == element }).length > 0
}

function fiveAscendingVowels(word) {
  var vowels = ["a","e","i","o","u"];
  var vowelsInWord = [];
  for (var i = 0; i < word.length; i++) {
    if (contains(vowels, word[i].toLowerCase())) {
      vowelsInWord.push(word[i]);
    }
  }
  for (var i = 0; i < vowels.length; i++) {
    if (vowels[i] != vowelsInWord[i]) { return false }
  }
  return true;
}

function sixPlusAscendingAlphaOrder(word) {
  if (word.length < 6) {
    return false;
  } else {
    var c = word.split("").map(function(a) { return a.toLowerCase() });
    var sortedC = word.split("").map(function(a) { return a.toLowerCase() }).sort();
    for (var i in c) {
      if (c[i] != sortedC[i]) {
        return false;
      }
    }
    return true;
  }
}

function returnWordsForTestFn(dict,testfn) {
  var words = fs.readFileSync(dict).toString().split("\n");
  var results = [];
  for (var i in words) {
    if (testfn(words[i])) {
      results.push(words[i]);
    }
  }
  return results;
}

var d = "unixdict.txt"
console.log(returnWordsForTestFn(d,fiveAscendingVowels));
console.log(returnWordsForTestFn(d,sixPlusAscendingAlphaOrder));

File: two_word_games.swift
#!/usr/bin/env swift

import Foundation

func fiveAscendingVowels(word: String) -> Bool {
  let vowels:[String] = ["a","e","i","o","u"]
  var vowelsInWord:[String] = word.characters.map { String($0) }.filter { vowels.contains($0) }
  if vowelsInWord.count < 5 { return false }
  for i in 0..<vowelsInWord.count {
    if vowelsInWord[i] != vowels[i] { return false }
  }
  return true
}

func sixPlusAscendingAlphaOrder(word: String) -> Bool {
  let c = word.characters.map { String($0) }
  if c.count < 6 { return false }
  let sortedC = c.sort { $0 < $1 }
  return sortedC == c
}

func findWordsForTest(dict: String, test: (String) -> Bool) -> [String] {
  let fileContents = try! String(contentsOfFile: dict, encoding: NSUTF8StringEncoding)
  let wordsArray = fileContents.componentsSeparatedByString("\n")
  return wordsArray.filter { test($0) }
}

print(findWordsForTest("unixdict.txt", test: fiveAscendingVowels))
print(findWordsForTest("unixdict.txt", test: sixPlusAscendingAlphaOrder))
